resources/install.rb

```diff
@@ -102,7 +112,9 @@
          description: 'Hash of additional sysconfig parameters to apply to the system'
 
 action :install do
-  package [apache_pkg, perl_pkg]
+  package [new_resource.apache_pkg, perl_pkg] do
+    version [new_resource.apache_version, nil] unless new_resource.apache_version.empty?
```

Comment:
I discovered an issue with this.  I believe the apache_version must also be applied for any module that must also install a package.

Example: `apache2_module 'ssl'`
This module installs the `mod_ssl` package and will upgrade Apache to the latest available.

Reply:
Good catch, thanks. Apologies for not testing it well enough.
I get the impression that it would be best solved with the use of attributes, but you don't use them here for some reason.
Alternatively installing everything in one place could be viable somehow,  the `package` resource is not used in many places, so perhaps refactoring the resources using it is a solution. Passing a property to them is also a way, I'm not a fan of that though, as it puts too much of a burden on an user and is easy to forget about and thus error-prone.
These are just loose thoughts. Let me know your feedback. Do you see any other possibilities?

Reply:
I too am still learning the proper ways of handling the transition from recipe-based to resource-based cookbooks and I'm not sure of the best methods yet myself.  @damacus do you have any suggestions?

Reply:
Switching from recipe based to resource based in one way means we provide a consistent interface to the user of the cookbook. All tunables should be pushed up to the resource interface instead of allowing the user to reach into the cookbook and change things that we can't test. 

If we're adding in a resource property that can change multiple internal values, or change how something is (for example) installed. Then we should have a test for that.

Reply:
This looks like the correct way of pushing up defaults that we can override if needs be. And thankfully it also looks like you'e done an awesome job with renaming the library helper to `default_`, which helps maintainers in the future, who are looking for default values!

Reply:
I'll take care of this issue in another PR once this is merged